### ISSUE TITLE
Silent failure when indexing using tuples

### ISSUE THREAD

**Issue by SNMS95** [jax-ml/jax#34127]
### Description

I have the following  snippet of code, aiming to set a jax array's values based on a boolean tuple.
```
import jax
import jax.numpy as jnp
from functools import partial

@partial(jax.jit, static_argnums=(1,))  
def fn(x, problem):
    x = x.at[problem].set(0.0)
    return x

x = jnp.array([1.0, 2.0, 3.0, 4.0])
problem = (True, False, True, False)
y = fn(x, problem)
print(y)
```

- This does not raise any errors but the output is simply `x`, completely unaffected.
- If I convert the tuple to a `np.array`, everything works well but I have to do this conversion inside the function (Otherwise, I will get the non-hashable error)
- If I convert to a `jax.Array`, it throws an error since the values are not concrete [But since they are static they should not be traced right?]

Is this expected behavior? Is there any other way to keep jax/numpy arrays as static?

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.5.3
jaxlib: 0.5.3
numpy:  2.3.3
python: 3.11.12 | packaged by conda-forge | (main, Apr 10 2025, 22:18:52) [Clang 18.1.8 ]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Darwin', node='mac.home', release='24.6.0', version='Darwin Kernel Version 24.6.0: Wed Oct 15 21:09:41 PDT 2025; root:xnu-11417.140.69.703.14~1/RELEASE_ARM64_T8122', machine='arm64')
```

**Comment by jakevdp** [comment]
Believe it or not, this is working as expected! The issue is that when you index with a tuple of boolean values, you're doing multiple zero-dimensional boolean indices, which is different than a boolean array typically used for masking. For example:
```python
>>> x = jnp.array([1.0, 2.0, 3.0, 4.0])

>>> x[True, False, True, False]  # tuple of scalar boleans
Array([], shape=(0, 4), dtype=float32)

>>> mask = jnp.array([True, False, True, False])  # array of booleans
>>> x[mask]
Array([1., 3.], dtype=float32)
```
The behavior of scalar boolean indices is admittedly a bit counter-intuitive, but is well-defined (and JAX inherits this behavior from NumPy; see [NumPy: Boolean array indexing](https://numpy.org/doc/stable/user/basics.indexing.html#boolean-array-indexing)).

The code you're using corresponds to the first case, and every element of the zero-sized slice in your array is being set to the new value, which of course leaves the original array unchanged. If you want boolean masking, you'll need to use an array rather than a tuple:
```python
>>> index = jnp.array([True, False, True, False])
>>> x.at[index].set(0)
Array([0., 2., 0., 4.], dtype=float32)
```
Now, if you switch to this, you'll immediately run into a problem because boolean mask updates cannot be done within JIT, because the size of the updates is dynamically shaped (see [JAX Sharp Bits: Dynamic Shapes](https://docs.jax.dev/en/latest/notebooks/Common_Gotchas_in_JAX.html#dynamic-shapes)). In this particular case (replacing masked elements with a scalar) you can get around this by instead writing the following:
```python
>>> index = jnp.array([True, False, True, False])
>>> jnp.where(index, 0, x)
Array([0., 2., 0., 4.], dtype=float32)
```
That computes the same result in a way that will be compatible with JIT compilation.
